Library/Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,70 @@
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.9.6.tar.gz"
+  sha256 "cf2dad62e50c5c5cff88a4ad959af3aa402aa7e8aa1692bdf807052c0fe4aeb9"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  ALL_CLIENTS = [
+    "access-management", "acm", "apigateway", "autoscaling",
+    "cloudformation", "cloudfront", "cloudhsm", "cloudsearch", "cloudtrail",
+    "codecommit", "codedeploy", "codepipeline", "cognito-identity",
+    "cognito-sync", "config", "datapipeline", "devicefarm", "directconnect",
+    "ds", "dynamobdb", "ec2", "ecr", "ecs", "elasticache", "elasticbeanstalk",
+    "elasticfilesystem", "elasticloadbalancing", "elasticmapreduce",
+    "elastictranscoder", "email", "es", "events", "firehose", "gamelift",
+    "glacier", "iam", "identity-management", "importexport", "inspector",
+    "iot", "kinesis", "kms", "lambda", "logs", "machinelearning",
+    "marketplacecommerceanalytics", "mobileanalytics", "monitoring",
+    "opsworks", "queues", "rds", "redshift", "route53", "s3", "sdb", "sns",
+    "sqs", "storagegateway", "sts", "support", "swf", "transfer", "waf",
+    "workspaces"].freeze
+
+  ALL_CLIENTS.each { |client| option "with-#{client}", "Build with #{client} client" }
+
+  option "with-custom-memory-management",
+         "Build with custom memory management"
+
+  option "with-static-linking", "Build with static linking"
+
+  depends_on "cmake" => :build
+
+  def install
+    clients = []
+    ALL_CLIENTS.each do |client|
+      clients << "aws-cpp-sdk-#{client}" if build.with? client
+    end
+
+    custom_memory_management = (build.with? "custom-memory-management") ? 1 : 0
+    static_linking = (build.with? "static-linking") ? 1 : 0
+
+    args = std_cmake_args
+    args << "-DCUSTOM_MEMORY_MANAGEMENT=#{custom_memory_management}"
+    args << "-STATIC_LINKING=#{static_linking}"
+    if clients.any?
+      args << "-DBUILD_ONLY=#{clients.join(";")}"
+    end
+
+    mkdir "build" do
+      system "cmake", *args, ".."
```

Comment:
This syntax is unsupported by Ruby 1.8, which we still support, for now. Should be:
```ruby
system "cmake", "..", *args
```
Instead.